applications/nrf_desktop/src/hw_interface/motion_simulated.c

```diff
@@ -72,12 +72,22 @@ static void motion_event_send(int16_t dx, int16_t dy)
 	APP_EVENT_SUBMIT(event);
 }
 
+static int64_t get_timestamp(void)
```

Comment:
Or maybe we could divide it by period for whole shape here to avoid overflows later on? (that could also allow us to properly handle overflows for 32 bit value here). Otherwise we may need to use bigger data types to prevent overflows later on.

Reply:
I was also thinking a bit about possible overflows during calculations in `generate_motion_event` (we now use value with more bits for timestamps). Took a closer look at the implementation - it seems that modulo operation would prevent overflows later on.

Reply:
> btw. Why not `uint64_t`? Both underlying APIs rely on unsigned values

you are right, changed to `uint64_t`.



> I was also thinking a bit about possible overflows during calculations in `generate_motion_event` (we now use value with more bits for timestamps). Took a closer look at the implementation - it seems that modulo operation would prevent these overflows for bigger timestamps.
> 
> Only the system's HW clock may overflow - optionally we could improve the implementation by handling HW clock overflow in this function and make the `get_timestamp` function it return value from 0 to time of drawing whole trajectory (so that we could keep trajectory continuous also after a clock overflow).

IMO the system's HW clock overflow is a nit corner case here. 
Theoretically, if my calculations are correct, it could happen after approx. 36h of continous module work (assuming `SYS_CLOCK_HW_CYCLES_PER_SEC` is 32kHz and timestamp is in `uint32_t`).
Moreover, it possibly would be just one jump in trajectory.
I would desist from any other changes here.

Reply:
I would not insist. Let's keep it as is then.